contracts/liquidity_hub/vault-network/vault/src/execute/deposit.rs

```diff
@@ -15,6 +15,12 @@ pub fn deposit(deps: DepsMut, env: Env, info: MessageInfo, amount: Uint128) -> S
         return Err(VaultError::DepositsDisabled {});
     }
 
+    // check that we are not currently in a flash-loan
+    if LOAN_COUNTER.load(deps.storage)? != 0 {
```

Comment:
Another check we could have here is if the vault is depositing into itself. The vault should not be allowed to flashloan against itself and then deposit the proceeds back into the vault. Or should it ? 
Also something to consider is if we have 2 vaults can we just repeat the problem using the vaults against one another

Reply:
I believe that's not possible unless we have code that explicitly does this from within the contract. If someone takes a flashloan and tries to deposit it back to the vault or so, the `info.sender` shouldn't be the vault but the user.

Reply:
This check disables all deposits during a flash-loan, so the situation you describe is impossible. But as @kerber0x says, we send the funds back to the user, rather than executing messages that they want. This eliminates a lot of security problems as you describe.

Reply:
Great, thank you for response ser